### PR TITLE
Implemented data download callback

### PR DESCRIPTION
**Pull request by Ashok28**
This adds possibility to further manipulate downloaded raw image data. Can be used i.e. for decrypting images.

### CODE REVIEW

File: Sources/ImageDownloader.swift [hunk 1]
```diff
@@ -141,6 +153,9 @@ extension ImageDownloaderDelegate {
     public func isValidStatusCode(_ code: Int, for downloader: ImageDownloader) -> Bool {
         return (200..<400).contains(code)
     }
+    public func imageDownloader(_ downloader: ImageDownloader, didDownload data: Data, for url: URL) -> Data? {
+        return nil
```
Comment (onevcat) [inline Sources/ImageDownloader.swift]: Maybe we should just return the original data instead of `nil`? Any reason to return a `nil` here?

**Comment by onevcat** [comment]
@Ashok28 Thanks for the commit. It could be useful and please take a look at the review comment above. Thank you!

File: Sources/ImageDownloader.swift (hunk 1 above)
Comment (Ashok28) [reply Sources/ImageDownloader.swift]: You're right, there's no particular reason, to return nil. I've pushed updated version.

**Comment by onevcat** [comment]
And I guess it would be better to interrupt the process and call completion handler with an error if `nil` is returned in this delegate method. By this, the user could have more control on what should happen according. (Say, not showing an image and getting an error when decrypting is failing).

@Ashok28  How do you think about that?

Anyway, I will merge this for now and do some improvement based on it. Thanks!

**Comment by Ashok28** [comment]
Definitely, this should have acted just as basic implementation, which I used for one specific use case, wasn't sure, if it's going to get merged. I can further improve it.

**Pull request merged**